File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/shape/Shape.java
package ja.burhanrashid52.photoeditor.shape;

import android.graphics.Canvas;
import android.graphics.Paint;

public interface Shape {

    void draw(Canvas canvas, Paint paint);
    void startShape(float x, float y);
    void moveShape(float x, float y);
    void stopShape();

}

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/shape/Shape.kt
package ja.burhanrashid52.photoeditor.shape

import android.graphics.Canvas
import android.graphics.Paint

interface Shape {
    fun draw(canvas: Canvas, paint: Paint)
    fun startShape(x: Float, y: Float)
    fun moveShape(x: Float, y: Float)
    fun stopShape()
}